src/lib/swapper/swappers/CowSwapper/cowBuildTrade/cowBuildTrade.ts

```diff
@@ -42,42 +53,50 @@ export async function cowBuildTrade(
 
   const sellAmountBeforeFeesCryptoBaseUnit = input.sellAmountBeforeFeesCryptoBaseUnit
 
-  const { assetReference: sellAssetErc20Address, assetNamespace: sellAssetNamespace } = fromAssetId(
-    sellAsset.assetId,
-  )
-
-  const { assetReference: buyAssetErc20Address, chainId: buyAssetChainId } = fromAssetId(
+  const {
+    assetReference: sellAssetAddress,
+    assetNamespace: sellAssetNamespace,
+    chainId: sellAssetChainId,
+  } = fromAssetId(sellAsset.assetId)
+  const { assetReference: buyAssetAddress, chainId: buyAssetChainId } = fromAssetId(
     buyAsset.assetId,
   )
 
   if (sellAssetNamespace !== 'erc20') {
     return Err(
       makeSwapErrorRight({
-        message: '[cowBuildTrade] - Sell asset needs to be ERC-20 to use CowSwap',
+        message: `[cowBuildTrade] - Sell asset needs to be ERC-20 to use CowSwap`,
         code: SwapErrorType.UNSUPPORTED_PAIR,
-        details: { sellAssetNamespace },
+        details: { sellAssetNamespace, sellAssetChainId },
       }),
     )
   }
 
-  if (buyAssetChainId !== KnownChainIds.EthereumMainnet) {
+  if (
+    !(
+      isCowswapSupportedChainId(buyAssetChainId, supportedChainIds) &&
+      buyAssetChainId === sellAssetChainId
+    )
+  ) {
     return Err(
       makeSwapErrorRight({
-        message: '[cowBuildTrade] - Buy asset needs to be on ETH mainnet to use CowSwap',
+        message: `[cowBuildTrade] - Both assets need to be on a network supported by CowSwap`,
         code: SwapErrorType.UNSUPPORTED_PAIR,
         details: { buyAssetChainId },
       }),
     )
   }
 
-  const buyToken =
-    buyAsset.assetId !== ethAssetId ? buyAssetErc20Address : COW_SWAP_ETH_MARKER_ADDRESS
+  const buyToken = !isNativeEvmAsset(buyAsset.assetId)
+    ? buyAssetAddress
+    : COW_SWAP_ETH_MARKER_ADDRESS
```

Comment:
Revert this to the previously correct terminology
https://github.com/shapeshift/web/pull/4595/files#r1209396771